### ISSUE TITLE
Playback fails in demo app after selecting all the tracks from the same audio track group

### ISSUE THREAD

**Issue by GouravSna** [google/ExoPlayer#5971]
**Issue:** Playback fails when we try to select all the tracks available in an audio group.

**Media:** https://bitdash-a.akamaihd.net/content/sintel/hls/playlist.m3u8 

**Crash Log:**

```
2019-05-30 14:15:49.249 12743-12743/com.google.android.exoplayer2.demo E/EventLogger: playerFailed [14.12, 16.62, window=0, period=0]
    com.google.android.exoplayer2.ExoPlaybackException: java.lang.ArrayIndexOutOfBoundsException: length=4; index=4
        at com.google.android.exoplayer2.ExoPlayerImplInternal.handleMessage(ExoPlayerImplInternal.java:397)
        at android.os.Handler.dispatchMessage(Handler.java:102)
        at android.os.Looper.loop(Looper.java:193)
        at android.os.HandlerThread.run(HandlerThread.java:65)
     Caused by: java.lang.ArrayIndexOutOfBoundsException: length=4; index=4
        at com.google.android.exoplayer2.trackselection.AdaptiveTrackSelection.setCheckpointValues(AdaptiveTrackSelection.java:810)
        at com.google.android.exoplayer2.trackselection.AdaptiveTrackSelection.getAllocationCheckpoints(AdaptiveTrackSelection.java:745)
        at com.google.android.exoplayer2.trackselection.AdaptiveTrackSelection.access$000(AdaptiveTrackSelection.java:37)
        at com.google.android.exoplayer2.trackselection.AdaptiveTrackSelection$Factory.createTrackSelections(AdaptiveTrackSelection.java:275)
        at com.google.android.exoplayer2.trackselection.DefaultTrackSelector.selectTracks(DefaultTrackSelector.java:1451)
        at com.google.android.exoplayer2.trackselection.MappingTrackSelector.selectTracks(MappingTrackSelector.java:400)
        at com.google.android.exoplayer2.MediaPeriodHolder.selectTracks(MediaPeriodHolder.java:209)
        at com.google.android.exoplayer2.ExoPlayerImplInternal.reselectTracksInternal(ExoPlayerImplInternal.java:1100)
        at com.google.android.exoplayer2.ExoPlayerImplInternal.handleMessage(ExoPlayerImplInternal.java:357)
        at android.os.Handler.dispatchMessage(Handler.java:102) 
        at android.os.Looper.loop(Looper.java:193) 
        at android.os.HandlerThread.run(HandlerThread.java:65)
```

**My Assumption:** I am assuming that when I select all the tracks available from an audio group it will behave like adaptive (track Index = -1).

**Question:**
If I check the stream it has the following audio track group,

```
#EXT-X-MEDIA:TYPE=AUDIO,GROUP-ID="stereo",LANGUAGE="en",NAME="English",DEFAULT=YES,AUTOSELECT=YES,URI="audio/stereo/en/128kbit.m3u8"
#EXT-X-MEDIA:TYPE=AUDIO,GROUP-ID="stereo",LANGUAGE="dubbing",NAME="Dubbing",DEFAULT=NO,AUTOSELECT=YES,URI="audio/stereo/none/128kbit.m3u8"

#EXT-X-MEDIA:TYPE=AUDIO,GROUP-ID="surround",LANGUAGE="en",NAME="English",DEFAULT=YES,AUTOSELECT=YES,URI="audio/surround/en/320kbit.m3u8"
#EXT-X-MEDIA:TYPE=AUDIO,GROUP-ID="surround",LANGUAGE="dubbing",NAME="Dubbing",DEFAULT=NO,AUTOSELECT=YES,URI="audio/stereo/none/128kbit.m3u8"
```
If I check the tracks in code then audio groups will be

1. English(stereo)/English(surround)   Unique Id:  1,0,0 /  1,0,1
2. dubbing(stereo)/dubbing(surround)    Unique Id:  1,1,0 /  1,1,1

How can we select auto for Audio ? 

There could be following an option to select auto from English group (1,0,-1) or dubbing group (1,1,-1). Can we do this ? I tried but the app crashed.

How ExoPlayer is selecting the audio in case if I select auto ?

**Comment by giladna** [comment]
@ojw28 
in previous version we were able to set adaptiveness in each of the groups since each audio group is working on its own language.

as for now the auto possibility is not available any more for each group
the questions is actually how adaptiveness for now is working across languages on latest version

**Comment by tonihei** [comment]
The reasons for this bug is that all the audio formats have no bitrate specified. We can and should make our algorithm more robust to gracefully handle this case, but I'm unsure what you are actually trying to achieve? There is no sensible way in which we can switch dynamically between two formats if they don't have bitrate information to guide the decision.

**Comment by tonihei** [comment]
@giladna I don't understand how your comment is related to the issue above. Can you rephrase it or open a new issue/question if unrelated?

**Comment by giladna** [comment]
@tonihei 
I was asking regarding auto bitrate selection in such case that 2 different languages are defined in 2 different groups

would it be possible to set auto bitrate selection for each group or auto bitrate selection is single option for a all audio tracks

**Comment by giladna** [comment]
@ojw28 @tonihei 

I am not sure in 100% but in m3u8 it is not expected to expose the audio bitrate  compare to dash where adaptation set expose it.
so I guess stream is fine

**Comment by giladna** [comment]
@ojw28  @tonihei 

one more question is if auto bitrate selection in m3u8  is relevant at all in multi language case?
since it is not straight forward compare to dash

**Comment by giladna** [comment]
![Screen Shot 2019-05-30 at 4 47 24 PM](https://user-images.githubusercontent.com/16818804/58637215-b44eaa80-82fa-11e9-8f92-bb978d01822b.png)
![Screen Shot 2019-05-30 at 4 45 47 PM](https://user-images.githubusercontent.com/16818804/58637216-b4e74100-82fa-11e9-82c4-485025e79324.png)


Attached formats for same stream but once in m3u8 and once in dash formats
it can be noticed that the bitrates are not reported in the m3u8 case

**Comment by tonihei** [comment]
Yes, HLS streams mostly don't have specific audio bitrates as the provided bitrates are only for combined  video+audio.

**Comment by tonihei** [comment]
Fixed by commit above.

**Comment by giladna** [comment]
@tonihei  thanks!

Regarding selecting auto bitrate for specific language in dash
how would it be applicable using ParametersBuilder because changing to auto bitrate selection in multiple languages is done by sending empty selection overrride which can be used only for one language in that form

![Screen Shot 2019-05-30 at 5 41 04 PM](https://user-images.githubusercontent.com/16818804/58640642-31315280-8302-11e9-9521-1f70f3747e82.png)
